daemon/k8s/pods.go

```diff
@@ -0,0 +1,135 @@
+	"k8s.io/apimachinery/pkg/util/duration"
+
+	"github.com/cilium/cilium/pkg/k8s"
+	"github.com/cilium/cilium/pkg/k8s/client"
+	slim_corev1 "github.com/cilium/cilium/pkg/k8s/slim/k8s/api/core/v1"
+	"github.com/cilium/cilium/pkg/k8s/utils"
+	nodeTypes "github.com/cilium/cilium/pkg/node/types"
+	"github.com/cilium/cilium/pkg/time"
+)
+
+// LocalPod is Cilium's internal model of the pods running on this node.
+type LocalPod struct {
+	*slim_corev1.Pod
+
+	// UpdatedAt is the time when [LocalPod] was last updated, e.g. it
+	// shows when the pod change was received from the api-server.
+	UpdatedAt time.Time `json:"updatedAt" yaml:"updatedAt"`
+}
+
+func (p LocalPod) TableHeader() []string {
+	return []string{
+		"Name",
+		"HostNetwork",
+		"PodIPs",
+		"Containers",
+		"Phase",
+		"Age",
+	}
+}
+
+func (p LocalPod) TableRow() []string {
+	var podIPs []string
+	for _, ip := range p.Status.PodIPs {
+		podIPs = append(podIPs, ip.IP)
+	}
+	var containers []string
+	for _, cont := range p.Spec.Containers {
+		var ports []string
+		for _, port := range cont.Ports {
+			if port.Name != "" {
+				ports = append(ports, fmt.Sprintf("%d/%s (%s)", port.ContainerPort, string(port.Protocol), port.Name))
+			} else {
+				ports = append(ports, fmt.Sprintf("%d/%s", port.ContainerPort, string(port.Protocol)))
+			}
+		}
+		contName := cont.Name
+		if len(ports) > 0 {
+			contName += " (" + strings.Join(ports, ",") + ")"
+		}
+		containers = append(containers, contName)
+	}
+	return []string{
+		p.Namespace + "/" + p.Name,
+		strconv.FormatBool(p.Spec.HostNetwork),
+		strings.Join(podIPs, ", "),
+		strings.Join(containers, ", "),
+		string(p.Status.Phase),
+		duration.HumanDuration(time.Since(p.UpdatedAt)),
+	}
+}
+
+const (
+	PodTableName = "k8s-pods"
+)
+
+var (
+	PodNameIndex = newNameIndex[LocalPod]()
+
+	PodTableCell = cell.Group(
+		cell.ProvidePrivate(NewPodTable),
+		cell.Provide(statedb.RWTable[LocalPod].ToTable),
+		cell.Invoke(registerPodReflector),
+	)
+)
+
+func PodByName(namespace, name string) statedb.Query[LocalPod] {
+	return PodNameIndex.Query(namespace + "/" + name)
+}
+
+func NewPodTable(db *statedb.DB) (statedb.RWTable[LocalPod], error) {
+	tbl, err := statedb.NewTable(
+		PodTableName,
+		PodNameIndex,
+	)
+	if err != nil {
+		return nil, err
+	}
+	return tbl, db.RegisterTable(tbl)
+}
+
+func registerPodReflector(jg job.Group, db *statedb.DB, cs client.Clientset, pods statedb.RWTable[LocalPod]) error {
+	if !cs.IsEnabled() {
+		return nil
+	}
+	cfg := podReflectorConfig(cs, pods)
+	return k8s.RegisterReflector(jg, db, cfg)
+}
+
+func podReflectorConfig(cs client.Clientset, pods statedb.RWTable[LocalPod]) k8s.ReflectorConfig[LocalPod] {
+	lw := utils.ListerWatcherWithModifiers(
+		utils.ListerWatcherFromTyped(cs.Slim().CoreV1().Pods("")),
+		func(opts *metav1.ListOptions) {
+			opts.FieldSelector = fields.ParseSelectorOrDie("spec.nodeName=" + nodeTypes.GetName()).String()
+		})
+	return k8s.ReflectorConfig[LocalPod]{
+		Name:          "daemon",
```

Comment:
I'd use `localPods` for the reflector name.

Reply:
Made it "daemon-k8s" and put it into "reflectorName" constant that should be used by all tables in `daemon/k8s`